### DISCUSSION TITLE
Versione Core e altre

### DISCUSSION THREAD

**Discussion by provenza** [indomus/forum#358] [GENERALE - Discussioni generiche]
Una domanda ho installato sul Raspberry 3 una versione core di Home Assistant su Python / Venv.
La versione installata è la 2021.1.5, guardando il sito home assistant vedo che la versione in uso è la 2021.5.5, ma lanciando l'upgrade non mi installa niente e mi diche che le dipendenze sono già aggiornate.
Sapete come mai?

`Requirement already satisfied, skipping upgrade: typing-extensions>=3.6.5 in /srv/homeassistant/lib/python3.7/site-pac
kages (from aiohttp==3.7.3->homeassistant) (3.7.4.3)                                                                  
Requirement already satisfied, skipping upgrade: chardet<4.0,>=2.0 in /srv/homeassistant/lib/python3.7/site-packages (
from aiohttp==3.7.3->homeassistant) (3.0.4)                                                                           
Requirement already satisfied, skipping upgrade: rfc3986[idna2008]<2,>=1.3 in /srv/homeassistant/lib/python3.7/site-pa
ckages (from httpx==0.16.1->homeassistant) (1.4.0)                                                                    
Requirement already satisfied, skipping upgrade: sniffio in /srv/homeassistant/lib/python3.7/site-packages (from httpx
==0.16.1->homeassistant) (1.2.0)                                                                                      
Requirement already satisfied, skipping upgrade: httpcore==0.12.* in /srv/homeassistant/lib/python3.7/site-packages (f
rom httpx==0.16.1->homeassistant) (0.12.2)                                                                            
Requirement already satisfied, skipping upgrade: idna>=2.0 in /srv/homeassistant/lib/python3.7/site-packages (from yar
l==1.6.3->homeassistant) (2.10)                                                                                       
Requirement already satisfied, skipping upgrade: zipp>=0.5 in /srv/homeassistant/lib/python3.7/site-packages (from imp
ortlib-metadata==1.6.0; python_version < "3.8"->homeassistant) (3.2.0)                                                
Requirement already satisfied, skipping upgrade: urllib3<1.27,>=1.21.1 in /srv/homeassistant/lib/python3.7/site-packag
es (from requests==2.25.0->homeassistant) (1.26.4)                                                                    
Requirement already satisfied, skipping upgrade: cffi!=1.11.3,>=1.8 in /srv/homeassistant/lib/python3.7/site-packages 
(from cryptography==3.2->homeassistant) (1.14.3)                                                                      
Requirement already satisfied, skipping upgrade: six>=1.4.1 in /srv/homeassistant/lib/python3.7/site-packages (from cr
yptography==3.2->homeassistant) (1.15.0)                                                                              
Requirement already satisfied, skipping upgrade: h11==0.* in /srv/homeassistant/lib/python3.7/site-packages (from http
core==0.12.*->httpx==0.16.1->homeassistant) (0.12.0)                                                                  
Requirement already satisfied, skipping upgrade: pycparser in /srv/homeassistant/lib/python3.7/site-packages (from cff
i!=1.11.3,>=1.8->cryptography==3.2->homeassistant) (2.20) `

**Comment by snakuzzo** [comment]
Qui non mi pare ci sia evidenza di quello che hai scritto.
Posta tutto quello che stai facendo e cortesemente formatta bene/meglio l'output usando correttamente il markdown.

**Comment by provenza** [comment]
ci provo

`sudo systemctl stop home-assistant@homeassistant`
`sudo -u homeassistant -H -s`
`source /srv/homeassistant/bin/activate`
`pip3 install --upgrade homeassistant`

`Requirement already up-to-date: homeassistant in /srv/homeassistant/lib/python3.7/site-packages (2021.1.5)`
`Requirement already satisfied, skipping upgrade: python-slugify==4.0.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (4.0.1)`
`Requirement already satisfied, skipping upgrade: ciso8601==2.1.3 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (2.1.3)`
`Requirement already satisfied, skipping upgrade: async-timeout==3.0.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (3.0.1)`
`Requirement already satisfied, skipping upgrade: requests==2.25.0 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (2.25.0)`
`Requirement already satisfied, skipping upgrade: pyyaml==5.3.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (5.3.1)`
`Requirement already satisfied, skipping upgrade: astral==1.10.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (1.10.1)`
`Requirement already satisfied, skipping upgrade: bcrypt==3.1.7 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (3.1.7)`
`Requirement already satisfied, skipping upgrade: voluptuous-serialize==2.4.0 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (2.4.0)`
`Requirement already satisfied, skipping upgrade: ruamel.yaml==0.15.100 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (0.15.100)`
`Requirement already satisfied, skipping upgrade: certifi>=2020.6.20 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (2020.12.5)`
`Requirement already satisfied, skipping upgrade: httpx==0.16.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (0.16.1)`
`Requirement already satisfied, skipping upgrade: PyJWT==1.7.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (1.7.1)`
`Requirement already satisfied, skipping upgrade: cryptography==3.2 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (3.2)`
`Requirement already satisfied, skipping upgrade: importlib-metadata==1.6.0; python_version < "3.8" in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (1.6.0)`
`Requirement already satisfied, skipping upgrade: yarl==1.6.3 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (1.6.3)`
`Requirement already satisfied, skipping upgrade: voluptuous==0.12.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (0.12.1)`
`Requirement already satisfied, skipping upgrade: pytz>=2020.1 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (2020.5)`
`Requirement already satisfied, skipping upgrade: attrs==19.3.0 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (19.3.0)`
`Requirement already satisfied, skipping upgrade: aiohttp==3.7.3 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (3.7.3)`
`Requirement already satisfied, skipping upgrade: pip<20.3,>=8.0.3 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (18.1)`
`Requirement already satisfied, skipping upgrade: jinja2>=2.11.2 in /srv/homeassistant/lib/python3.7/site-packages (from homeassistant) (2.11.2)`
`Requirement already satisfied, skipping upgrade: text-unidecode>=1.3 in /srv/homeassistant/lib/python3.7/site-packages (from python-slugify==4.0.1->homeassistant) (1.3)`
`Requirement already satisfied, skipping upgrade: urllib3<1.27,>=1.21.1 in /srv/homeassistant/lib/python3.7/site-packages (from requests==2.25.0->homeassistant) (1.26.4)`
`Requirement already satisfied, skipping upgrade: idna<3,>=2.5 in /srv/homeassistant/lib/python3.7/site-packages (from requests==2.25.0->homeassistant) (2.10)`
`Requirement already satisfied, skipping upgrade: chardet<4,>=3.0.2 in /srv/homeassistant/lib/python3.7/site-packages (from requests==2.25.0->homeassistant) (3.0.4)`
`Requirement already satisfied, skipping upgrade: six>=1.4.1 in /srv/homeassistant/lib/python3.7/site-packages (from bcrypt==3.1.7->homeassistant) (1.15.0)`
`Requirement already satisfied, skipping upgrade: cffi>=1.1 in /srv/homeassistant/lib/python3.7/site-packages (from bcrypt==3.1.7->homeassistant) (1.14.3)`
`Requirement already satisfied, skipping upgrade: rfc3986[idna2008]<2,>=1.3 in /srv/homeassistant/lib/python3.7/site-packages (from httpx==0.16.1->homeassistant) (1.4.0)`
`Requirement already satisfied, skipping upgrade: httpcore==0.12.* in /srv/homeassistant/lib/python3.7/site-packages (from httpx==0.16.1->homeassistant) (0.12.2)`
`Requirement already satisfied, skipping upgrade: sniffio in /srv/homeassistant/lib/python3.7/site-packages (from httpx==0.16.1->homeassistant) (1.2.0)`
`Requirement already satisfied, skipping upgrade: zipp>=0.5 in /srv/homeassistant/lib/python3.7/site-packages (from importlib-metadata==1.6.0; python_version < "3.8"->homeassistant) (3.2.0)`
`Requirement already satisfied, skipping upgrade: multidict>=4.0 in /srv/homeassistant/lib/python3.7/site-packages (from yarl==1.6.3->homeassistant) (5.1.0)`
`Requirement already satisfied, skipping upgrade: typing-extensions>=3.7.4; python_version < "3.8" in /srv/homeassistant/lib/python3.7/site-packages (from yarl==1.6.3->homeassistant) (3.7.4.3)`
`Requirement already satisfied, skipping upgrade: MarkupSafe>=0.23 in /srv/homeassistant/lib/python3.7/site-packages (from jinja2>=2.11.2->homeassistant) (1.1.1)`
`Requirement already satisfied, skipping upgrade: pycparser in /srv/homeassistant/lib/python3.7/site-packages (from cffi>=1.1->bcrypt==3.1.7->homeassistant) (2.20)`
`Requirement already satisfied, skipping upgrade: h11==0.* in /srv/homeassistant/lib/python3.7/site-packages (from httpcore==0.12.*->httpx==0.16.1->homeassistant) (0.12.0)`

**Reply by snakuzzo** [reply]
Quando usi il markdown...seleziona TUTTO il blocco di codice e poi clicca sul pulsantino per evidenziare il codice (LEGGI [QUI](https://github.com/indomus/forum))

Tornando al tuo problema... che versione di python e pip hai ?

**Comment by provenza** [comment]
* Python 3.7.3
* pip 18.1 from /usr/lib/python3/dist-packages/pip (python 3.7)

**Reply by snakuzzo** [reply]
Non va bene.
Python 3.7 non è supportato più già da un pezzo e la versione di pip è datata.
Devi installare la 3.8 o 3.9 di Python e ricreare il venv

Leggi [QUESTA GUIDA](https://indomus.it/guide/aggiornare-python-su-raspbian-alla-3-8-x-o-superiori-e-ripristinare-home-assistant-core/)

**Comment by provenza** [comment]
purtroppo l'installazione di python si ferma sul comando
`make -j 8`

mi sa che devo ripartire da zero.

**Reply by snakuzzo** [reply]
No...perché RapOS mi sa che arriva (per ora) massimo alla 3.7.
Prova a vedere bene che errore hai. La guida è stata seguita da tante persone e questo blocco non mi risulta.
P.S.  Non rispondere sempre come nuovo thread...rispondi sotto la stessa discussione per favore!

**Reply by provenza** [reply]
ok.
In realtà non ricevo errori specifici. si ferma sul comando e se provo a dare un invio mi restituisce come messaggio **connection reset**

**Reply by snakuzzo** [reply]
Hai altri problemi. Quel connection reset non dipende dall'installazione che stai facendo.
Difficile dire da cosa possa dipendere solo con queste informazioni. Mi spiace.
Prova a guardare nel syslog`/var/log/syslog` cosa viene loggato quando ti succede...più di questo non posso dirti
